apps/prairielearn/src/pages/authLogin/authLogin.test.ts

```diff
@@ -0,0 +1,188 @@
+import { decode } from 'entities';
+import { afterAll, assert, beforeAll, describe, it } from 'vitest';
+
+import * as assets from '../../lib/assets.js';
+import { withConfig } from '../../tests/utils/config.js';
+
+import { AuthLoginInstitution } from './authLogin.html.js';
+
+describe('authLogin', () => {
+  // Bare minimum to be able to render the page programmatically.
+  beforeAll(() => assets.init());
+  afterAll(() => assets.close());
```

Comment:
I took a slightly different approach to testing here: since the page is a function that spits out HTML and the props are easy to specify inline, I'm just calling the function directly instead of making a full HTTP request to the page. This keeps the tests simple and importantly, really really fast. This probably isn't appropriate for every page but I think it works well here where we're just interested in making sure the rendering logic is working well.